File: 9-12-24/lazy.hs

numbers = [1..100]

odd' n = n `mod` 2 /= 0

getNodds n = take n (filter odd' numbers)

-- I like this version more than the other one
getFirstNOdds n = take n (filter odd [1..(n * 2)])

--------------------------------------------------

x :: Int -- "x has type int"

type point = (Double, Double)

File: 9-12-24/lazy.js
const numbers = Array.from(Array(100).keys(), (i) => i + 1);

// console.log(numbers);

function odd(n) {
    return n % 2 != 0;
}

function getNOddNumbers(n) {
    return numbers.filter(odd).slice(0, n);
}

// console.log(getNOddNumbers(5));

function getFirstNOddNumbers(n) {
    return Array.from(Array(n * 2).keys(), (i) => {
        return i + 1;
    }).filter((m) => {
        return m % 2 == 1;
    }).slice(0, n);
};

console.log(getFirstNOddNumbers(100));